src/modules/service-catalog/components/useItemFormFields.tsx

```diff
@@ -71,6 +71,10 @@ const fetchTicketFields = async (
   const formData = await formResponse.json();
   const fieldsData = await fieldsResponse.json();
 
+  if (!formData.ticket_form.active || formData.ticket_form.id !== form_id) {
```

Comment:
Moving this convo here:

> > Wouldn't the fetch call to `/api/v2/ticket_forms/${form_id}` fail with 404 in this scenario?

> That is a good question. The issue here is that forms are soft deleted. So although the form is "deleted" it will still be available and will be fetched but with the active property set to false.
> That is why I added a check for the active field as well. Hope this answers your question.

That makes sense for the first check: `formData.ticket_form.active`.
I'm trying to understand the second one: `formData.ticket_form.id !== form_id`.
Is there a case where a call to `/api/v2/ticket_forms/${form_id}` would return a different form id? 😕

Reply:
It'd be really really weird to do fetch for a form but get another 🤨 At this point nothing really surprises me though 😅 

Nonetheless, we should get to the bottom as these checks add quite a lot of distrust in the APIs and from my experience we'll never dare to remove them.

Reply:
I was checking and @Fredx87 is right. We fallback to other forms when we create a request with a form that is inactive/soft deleted. This check is not needed. We already handle the errors when the response is a 404, so we should cover the rest already. I'll update the code and the tests.

Reply:
I was able to test both scenarios (inactive form and missing form) and in fact we did not need that check. With the code as it is now, we should be able to handle both situations.

@luis-almeida and @Fredx87 can you re-review the PR? 🙏 
Also thank you both for the input.